### PR TITLE
[3006.x] Duplicate `arm64` repo paths for rpm-based distros to `aarch64` paths

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
See title

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/65198
See also: https://github.com/saltstack/salt/pull/65205

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by s0undt3ch — DISMISSED** [review]

File: tools/ci.py [hunk 1]
```diff
@@ -757,57 +739,103 @@ def pkg_matrix(
...
+    ]
+
+    for version, backend in adjusted_versions:
+        prefix = prefixes[backend]
+        # TODO: Remove this after 3009.0
+        if backend == "relenv" and version >= tools.utils.Version("3007.0"):
+            prefix.replace("/arm64/", "/aarch64/")
+        # Using a paginator allows us to list recursively and avoid the item limit
+        page_iterator = paginator.paginate(
+            Bucket="salt-project-prod-salt-artifacts-release", Prefix=prefix
```
Comment (s0undt3ch) [inline tools/ci.py:809]:
```suggestion
            Bucket=f"salt-project-{tools.utils.SPB_ENVIRONMENT}-salt-artifacts-release", Prefix=prefix
```

**Comment by MKLeb** [comment]
New passing staging run: https://github.com/MKLeb/salt/actions/runs/6867077920

**Comment by MKLeb** [comment]
Final passing staging run: https://github.com/MKLeb/salt/actions/runs/6880738359

File: tools/ci.py [hunk 2]
```diff
@@ -797,78 +765,103 @@ def pkg_matrix(
...
+            )
+            for session in ("upgrade", "downgrade"):
+                _matrix.append(
+                    {
+                        "test-chunk": f"{session}-classic"
+                        if backend == "classic"
+                        else session,
+                        "version": str(version),
+                    }
+                )
```
Comment (s0undt3ch) [inline tools/ci.py:862]:
```suggestion
                if backend == "classic":
                    session += "-classic"
                _matrix.append(
                    {
                        "test-chunk": session,
                        "version": str(version),
                    }
                )
                if backend == "relenv" and fips is True and distro_slug.startswith(("photonos-4", "photonos-5")):
                    # Repeat the last one, but with fips
                    _matrix.append({"fips": "fips", **_matrix[-1]})
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by MKLeb** [comment]
https://github.com/MKLeb/salt/actions/runs/6895006120

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**